scripts/dts/gen_defines.py

```diff
@@ -599,6 +605,55 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
+def write_dmas(node):
+    if "dmas" not in node.props:
+        return
+
+    write_phandle_val_list(node.props["dmas"])
+
+    # Client parameter syntax may vary with the DMA controller changes.
+    for dma_i, dma in enumerate(node.props["dmas"].val):
+        # Only this controller currently generates extra #defines
+        if "st,stm32-dma" in dma.controller.compats:
+            # One common argument 'channel' and 3 private arguments for stm32
+            # dma controller.
+            if "slot" not in dma.data:
+                err("Missing 'slot' in DMA consumer node {}.", node.reg)
```

Comment:
Probably want `{!r}` (the short `__repr__()` representation) and `dma.controller` instead of `node.reg`.